crates/sui-core/src/test_utils.rs

```diff
@@ -105,8 +107,9 @@ pub fn create_fake_transaction() -> Transaction {
 pub fn to_sender_signed_transaction(
     data: TransactionData,
     signer: &dyn Signer<Signature>,
-) -> Transaction {
+) -> VerifiedTransaction {
     let signature = Signature::new_temp(&data.to_bytes(), signer);
     // let signature = Signature::new_secure(&data, Intent::default(), signer).unwrap();
-    Transaction::new(data, signature)
+    let tx = Transaction::new(data, signature);
+    VerifiedTransaction::new_unchecked(tx)
```

Comment:
We have a signature here, we should just call verify with `.unwrap()` instead of relying on `new_unchecked`. I know this is test code but reducing calls to `new_unchecked` where possible seems like a good goal

Reply:
I went back and forth on this, but can certainly change it. I am tempted to make a `Transaction::new_with_signer()` method that takes the signer and produces a VerifiedTransaction, but this PR is big enough.

Reply:
yeah that would be a good alternative, but yeah we can delay that to another pr